### PR TITLE
A slice for push operations

### PR DESCRIPTION
**Pull request by Sammers21**
Related to  #1 and #12

### CODE REVIEW

**Comment by Sammers21** [comment]
@g4s8, pay attention to the PR.

File: src/main/java/com/artipie/helm/HelmSlice.java [hunk 1]
```diff
@@ -49,6 +49,13 @@
     public HelmSlice(final Storage storage) {
         super(
             new SliceRoute(
+                new SliceRoute.Path(
+                    new RtRule.Multiple(
+                        new RtRule.ByMethod(RqMethod.POST),
```
Comment (g4s8) [inline src/main/java/com/artipie/helm/HelmSlice.java:55]: @Sammers21 are you sure we need to support `POST` and `PUT` in the same manner? `POST` request usually doesn't include resource id, but `PUT` does. Example:
 - `POST /helm/charts` - to upload a chart and get chart's id, e.g. `/helm/charts/somechart` (client doesn't know chart id)
 - `PUT /helm/charts/somechart` to upload `somechart` (client knows chart id)

File: src/main/java/com/artipie/helm/PushChartSlice.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import org.reactivestreams.Publisher;
+
+/**
+ * A Slice which accept archived charts, save them into a storage and trigger index.yml reindexing.
+ *
+ * @since 0.2
+ * @checkstyle MethodBodyCommentsCheck (500 lines)
+ */
+@SuppressWarnings({"PMD.UnusedPrivateField", "PMD.SingularField"})
+public final class PushChartSlice implements Slice {
```
Comment (g4s8) [inline src/main/java/com/artipie/helm/PushChartSlice.java:43]: @Sammers21 we need an integration test for this skeleton where you can describe the logic of upload from client side

File: src/main/java/com/artipie/helm/HelmSlice.java (hunk 1 above)
Comment (Sammers21) [reply src/main/java/com/artipie/helm/HelmSlice.java:55]: This is compatible both with curl:
```
curl --data-binary "@redis-10.5.7.tgz" http://localhost:1235/helm
```
(which sends a POST request) and help push plugin.

**Comment by Sammers21** [comment]
@g4s8 , review the PR once again please

**Comment by 0pdd** [comment]
@Sammers21 2 puzzles [#14](https://github.com/artipie/helm-adapter/issues/14), [#15](https://github.com/artipie/helm-adapter/issues/15) are still not solved.

**Comment by 0pdd** [comment]
@Sammers21 the puzzle [#14](https://github.com/artipie/helm-adapter/issues/14) is still not solved; solved: [#15](https://github.com/artipie/helm-adapter/issues/15).

**Pull request merged**